src/node.cc

```diff
@@ -2261,6 +2261,56 @@ static void Uptime(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+static void GetResourceUsage(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  uv_rusage_t rusage;
+  int err = uv_getrusage(&rusage);
+  if (err)
+    return env->ThrowUVException(err, "uv_getrusage");
+
+  Local<Object> resource_usage = Object::New(env->isolate());
```

Comment:
I wonder if it still faster to create the object in JS land and pass it to the C++ side. I think @trevnorris had done something like this before (I can't recall exactly what part of core that was). I know `Object::New()` in general is quite slow (even compared to say `Array::New()`) from my own experience with third-party addons, so I could definitely see it possibly being faster in JS land.